hibernate-core/src/main/java/org/hibernate/event/internal/DefaultLoadEventListener.java

```diff
@@ -309,7 +309,7 @@ private Object proxyOrLoad(
 			persistenceContext.addEntity(
 					entity,
 					Status.MANAGED,
-					ArrayHelper.filledArray( LazyPropertyInitializer.UNFETCHED_PROPERTY, Object.class, persister.getPropertyTypes().length ),
+					new Object[ persister.getPropertyTypes().length ],
```

Comment:
Why is this change necessary? Is it possible that a loaded property will be mistaken for null?